File: Dota2 Senate/dota2-senate.java
class Solution {
    public String predictPartyVictory(String senate) {
        Queue<Integer> r = new LinkedList<>();
        Queue<Integer> d = new LinkedList<>();
        int n = senate.length();

        for (int i = 0; i < n; i++) {
            if (senate.charAt(i) == 'R') {
                r.add(i);
            } else {
                d.add(i);
            }
        }

        while (!r.isEmpty() && !d.isEmpty()) {
            int rId = r.poll();
            int dId = d.poll();
            if (rId < dId) {
                r.add(rId + n);
            } else {
                d.add(dId + n);
            }
        }

        return r.size() > d.size() ? "Radiant" : "Dire";
    }
}

File: Dota2 Senate/dota2-senate.cpp
class Solution {
public:
    string predictPartyVictory(string senate) {
        queue<int> r;
        queue<int> d;
        int n = senate.length();
        for (int i = 0; i < n; i++) {
            senate[i] == 'R' ? r.push(i) : d.push(i);
        }
        while (!r.empty() && !d.empty()) {
            int r_id = r.front();
            int d_id = d.front();
            r.pop();
            d.pop();
            r_id < d_id ? r.push(r_id + n) : d.push(d_id + n);
        }
        return r.size() > d.size() ? "Radiant" : "Dire";
    }
};